contracts/governance/staking/StakedTokenBPT.sol

```diff
@@ -51,81 +55,75 @@ contract StakedTokenBPT is StakedToken {
 
     /**
      * @param _nexus System nexus
-     * @param _rewardsToken Token that is being distributed as a reward. eg MTA
-     * @param _stakedToken Core token that is staked and tracked (e.g. MTA)
+     * @param _rewardsToken Token that is being distributed as a reward. eg Meta (MTA)
+     * @param _questManager Centralised manager of quests
+     * @param _stakedToken Core token that is staked and tracked e.g. mStable MTA/WETH Staking BPT (mBPT)
      * @param _cooldownSeconds Seconds a user must wait after she initiates her cooldown before withdrawal is possible
-     * @param _unstakeWindow Window in which it is possible to withdraw, following the cooldown period
      * @param _bal Balancer addresses, [0] = $BAL addr, [1] = BAL vault
      * @param _poolId Balancer Pool identifier
+     * @param _balancerGauge Address of the Balancer Pool Token Gauge. eg mBPT Gauge (mBPT-gauge)
      */
     constructor(
         address _nexus,
         address _rewardsToken,
         address _questManager,
         address _stakedToken,
         uint256 _cooldownSeconds,
-        uint256 _unstakeWindow,
         address[2] memory _bal,
-        bytes32 _poolId
+        bytes32 _poolId,
+        address _balancerGauge
     )
         StakedToken(
             _nexus,
             _rewardsToken,
             _questManager,
             _stakedToken,
             _cooldownSeconds,
-            _unstakeWindow,
             true
         )
     {
         BAL = IERC20(_bal[0]);
         balancerVault = IBVault(_bal[1]);
         poolId = _poolId;
+        balancerGauge = IBalancerGauge(_balancerGauge);
     }
 
     /**
      * @param _nameArg Token name
      * @param _symbolArg Token symbol
      * @param _rewardsDistributorArg mStable Rewards Distributor
-     * @param _balRecipient contract that can redistribute the $BAL
      * @param _priceCoefficient Initial pricing coefficient
      */
     function initialize(
         bytes32 _nameArg,
         bytes32 _symbolArg,
         address _rewardsDistributorArg,
-        address _balRecipient,
         uint256 _priceCoefficient
-    ) external initializer {
-        __StakedToken_init(_nameArg, _symbolArg, _rewardsDistributorArg);
-        balRecipient = _balRecipient;
-        priceCoefficient = _priceCoefficient;
-    }
+    ) external {
```

Comment:
was the removal of  modifier `initializer` intentional ?

Reply:
I think that i get it,  the init method can be called again, but the reward distributor can only be set once.

Reply:
Ideally, I would leave the `initialize` function with the `initializer` modifier but in order to get the contract size down I had to repurpose the `initialize` function to also do the migration.